Network/Socket/Types.hsc

```diff
@@ -1156,15 +1156,15 @@ unixPathMax = #const sizeof(((struct sockaddr_un *)NULL)->sun_path)
 -- | Write the given 'SockAddr' to the given memory location.
 pokeSockAddr :: Ptr a -> SockAddr -> IO ()
 pokeSockAddr p sa@(SockAddrUnix path) = do
-    when (length path > unixPathMax) $ error
+    let pathC = map castCharToCChar path
+    when (length pathC >= unixPathMax) $ error
       $ "pokeSockAddr: path is too long in SockAddrUnix " <> show path
       <> ", length " <> show (length path) <> ", unixPathMax " <> show unixPathMax
```

Comment:
It might be good to change the error message to `show (length pathC)`; currently it's equivalent since done with `map`, but maybe in the future we lift the constraint in the future that the `path` must not contain unicode chars (which is a pretty weird contraint because it should prevent anybody who uses non-ASCII chars in their home directory to place sockets there).

Otherwise, looks good!

Reply:
Fixed. Please check it again.